interim-jan-2021/resolver-behavior-self-description.md

```diff
@@ -0,0 +1,27 @@
+# Resolver behavior self-description
+
+## Defining local-only namespaces
+* Express namespaces which only this resolver can resolve
+* Ex 1: Enterprise resolvers serving corporation-specific namespaces
+* Ex 2: Public Wi-Fi or cellular networks offering network-local services
+
+## Defining per-namespace optimization
+* Express namespaces for which this resolver provides preferable resolutions
+* Ex 1: ISP routes public content requests to network caches
+* Ex 2: Cellular provider charges less for content from specific servers
+* Ex 3: Public resolver designated to serve a namespace to limit parties privy to resolution
+
+## Defining answer modification behavior
+* Express how query answers are modified by the resolver (but not necessarily _why_)
+* Ex 1: Will not resolve names breaching resolver policy 
+    * May or may not define applicable namespaces
+* Ex 2: Will drop specific record types
```

Comment:
What's the use case for this?  RFC 3597 is pretty clear that this is not allowed, and I've never encountered a use case where it was needed.

Reply:
This isn't a scenario I'm happy to support, but I included it as there are bound to be servers who choose to do so anyway (say networks which forbid ECH and similar technologies). It would be useful for clients to know this is by design rather than see other issue they should investigate.

Reply:
I don't think we should support this.  I'm not aware of any present need for it, and it directly contravenes correct DNS behavior.

> networks which forbid ECH and similar technologies

It is not logical for a resolver to do this based on RR type.  If the operator wants to prevent access to a domain, it can do so directly from the QNAME, rather than trying to force the client to leak the name again (which is not even guaranteed to work, due to connection coalescing).

Reply:
I would prefer to make this a non-goal.

I note that the example given in that documentation is to refuse the "ANY" QTYPE, which is actually allowed (RFC 8482).  A flag for the "refuse-ANY" behavior could be interesting.  Perhaps a missing requirement here is extensibility, to allow adding more signals over time.  Then if the IETF someday decides that refusing MX queries is OK, we could add a flag for that.

Reply:
I updated your PR with the non-goal indication; resolving.